pkg/build/controller/build/build_controller.go

```diff
@@ -350,6 +351,16 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 		}
 	}
 
+	// set the pushSecret that will be needed by the build to push the image to the registry
+	// at the end of the build.
+	build.Spec.Output.PushSecret = secret
+
+	// ensure the build object the pod sees starts with a clean set of reasons/messages,
+	// rather than inheriting the potential "invalidoutputreference" message from the current
+	// build state.
+	build.Status.Reason = ""
```

Comment:
(At some point I think we discussed having the build pod do a GET to pick up the build object instead of serializing it into an env var.  That would solve this problem, assuming we had no cache/race condition issues, but for now i don't intend to take that on in this PR)